src/python/pants/option/global_options.py

```diff
@@ -51,6 +51,13 @@ def to_glob_match_error_behavior(self) -> GlobMatchErrorBehavior:
     return GlobMatchErrorBehavior(self.value)
 
 
+class OwnersNotFoundBehavior(Enum):
```

Comment:
`Owner` as a concept is easy enough to understand once explained, but it also something additional to the [first concepts](https://www.pantsbuild.org/first_concepts.html) that a new user would need to learn.

Would it make sense to have this be more verbose to better represent what's happening (something like `TargetsNotFoundForFileBehavior`). There may be other context I'm missing here with regards to this change / the history behind 'owners', but if we're considering dropping `--owners-of` it might be a good time to consider?